internal/tenant/query.go

```diff
@@ -0,0 +1,29 @@
+package tenant
+
+import (
+	"context"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/sourcegraph/zoekt/internal/tenant/internal/tenanttype"
+)
+
+// Match returns true if the path is a subdirectory of tenant's index directory.
+func Match(ctx context.Context, path string) bool {
+	if !EnforceTenant() {
+		return true
+	}
+
+	t, err := tenanttype.FromContext(ctx)
+	if err != nil {
+		return false
+	}
```

Comment:
could this lead to strange errors where a missing tnt in ctx will silently return zero results always? should we return the error instead?

Reply:
Good point. Now we return an error.